File: app/src/main/java/com/eye/cool/book/adapter/LayoutId.java
package com.eye.cool.book.adapter;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by ycb on 16/4/18.
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface LayoutId {

  int value() default 0;
}

File: app/src/main/java/com/eye/cool/book/adapter/LayoutId.kt
package com.eye.cool.book.adapter

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

/**
 * Created by ycb on 16/4/18.
 */
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
@Retention(RetentionPolicy.RUNTIME)
annotation class LayoutId(val value: Int = 0)